src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterUri.cs

```diff
@@ -27,27 +22,13 @@ public override Uri Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSeri
         public override void Write(Utf8JsonWriter writer, Uri value, JsonSerializerOptions options)
         {
             // TODO: remove preprocessing when https://github.com/dotnet/corefx/issues/38647 is implemented.
-            string uriString = value.OriginalString;
-
-            if (uriString == string.Empty)
-            {
-                uriString = value.ToString();
-            }
-
-            writer.WriteStringValue(uriString);
+            writer.WriteStringValue(value.OriginalString);
```

Comment:
This will prevent relative URIs: ["This instance represents a relative URI, and this property is valid only for absolute URIs."](https://docs.microsoft.com/en-us/dotnet/api/system.uri.originalstring?view=netframework-4.8)

(I didn't realize how complicated the URI object was w.r.t. absolute vs relative URI's until just now.)

Reply:
`OriginalString` seems to work for both relative and absolute -

```
Uri.TryCreate("https://domain/path", UriKind.Absolute, out Uri uri1);
Console.WriteLine(uri1.OriginalString); // https://domain/path
Console.WriteLine(uri1.ToString()); // https://domain/path
		
Uri.TryCreate("~/path", UriKind.Relative, out Uri uri2);
Console.WriteLine(uri2.OriginalString); // ~/path
Console.WriteLine(uri2.ToString()); // ~/path
```

Per documentation - `ToString()` seems the safest 
(no exceptions) for now till this support is properly implemented in the reader/writer - https://github.com/dotnet/corefx/issues/38647.